flow/connectors/postgres/cdc.go

```diff
@@ -362,6 +362,10 @@ func PullCdcRecords[Items model.Items](
 
 	for {
 		if pkmRequiresResponse {
+			if cdcRecordsStorage.IsEmpty() {
+				req.ConsumedOffset.Store(int64(clientXLogPos))
```

Comment:
ConsumedOffset must only be updated after catalog updated, otherwise pg may remove wal & restart will request old value in catalog

Reply:
makes sense